programs/disks/disks.cpp

```diff
@@ -0,0 +1,354 @@
+}
+
+void DisksFunc::listDisks()
+{
+    while(argumentsToDo[0] != "list-disks")
+        argumentsToDo.erase(argumentsToDo.begin());
+
+    argumentsToDo.erase(argumentsToDo.begin());
+
+    if (int(argumentsToDo.size()) == 0)
+    {
+        std::vector<std::string> disksNames;
+
+        for (auto & [disk_name, disk] : global_context->getDisksMap())
+        {
+            disksNames.push_back(disk->getName());
+        }
+
+        std::cout << "Disks names: \n";
+        for (auto it = disksNames.begin(); it != disksNames.end(); it++)
+        {
+            std::cout << *it << "\n";
+        }
+    }
+    else
+    if (int(argumentsToDo.size()) > 0)
+    {
+        curCommand = "list-disks";
+        init(argumentsToDo);
+        run();
+    }
+}
+
+void DisksFunc::listFiles()
+{
+    if (curCommand == "start")
+    {
+        if (int(argumentsToDo.size()) == 1)
+        {
+            printHelpMessage();
+            throw DB::Exception("Bad Arguments", DB::ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        while (argumentsToDo[0] != "ls")
+            argumentsToDo.erase(argumentsToDo.begin());
+
+        argumentsToDo.erase(argumentsToDo.begin());
+
+        curCommand = "ls";
+        init(argumentsToDo);
+        run();
+    }
+    else
+        if (curCommand == "ls")
+        {
+            if (int(argumentsToDo.size()) > 1)
+            {
+                printHelpMessage();
+                throw DB::Exception("Bad Arguments", DB::ErrorCodes::BAD_ARGUMENTS);
+            }
+
+            diskPath.parsPath(in[0]);
+
+            std::vector<std::string> file_names;
+
+            DiskPtr disk_test = global_context->getDisk(diskPath.getDiskName());
+
+            std::string full_path = disk_test->getPath();
+
+            if (diskPath.getDiskName() == "default")
+                full_path = "/var/lib/clickhouse/data/default/";
+
+            full_path.pop_back();
+            full_path += diskPath.getPath();
+
+            disk_test->listFiles(full_path, file_names);
+
+            for (auto iter = file_names.begin(); iter != file_names.end(); iter++)
+            {
+                std::cout << *iter << "\n";
+            }
+        }
+}
+
+void DisksFunc::reloadConfiguration()
+{
+    config_path = config().getString("config-file", getDefaultConfigFileName());
+    DB::ConfigProcessor config_processor(config_path, false, true);
+    config_processor.setConfigPath(fs::path(config_path).parent_path());
+    loaded_config = config_processor.loadConfig();
+    if (last_configuration != nullptr)
+        config().removeConfiguration(last_configuration);
+    last_configuration = loaded_config.configuration.duplicate();
+    config().add(last_configuration, PRIO_DEFAULT, false);
+}
+
+void DisksFunc::initialize(Poco::Util::Application & self)
+{
+    Poco::Util::Application::initialize(self);
+    if (curCommand == "start")
+        reloadConfiguration();
+}
+
+void DisksFunc::printHelpMessage()
+{
+    if (curCommand == "start")
```

Comment:
Maybe we should build spcial command description object, that contains info about command name, method to call, program options, so you can generate help message in generic way here based only on that information.
It would simplify adding new commands, by add new item in commands map instead of searching a lot of methods where you should put new **if** statement